gcc/rust/ast/rust-ast.h

```diff
@@ -876,7 +874,7 @@ class MetaListNameValueStr;
 
 /* Base statement abstract class. Note that most "statements" are not allowed in
  * top-level module scope - only a subclass of statements called "items" are. */
-class Stmt : public Node
+class Stmt : public Node, public Visitable
```

Comment:
Wouldn't it be easier for `Node` to inherit from `Visitable`? This way you don't have to tweak Expr, Stmt, Type, and maybe more classes. It also makes sense semantically IMO

Reply:
Hmm maybe this could save a few lines and be more correct yes. I'll give it a try :)